java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/Application.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.runtimes.builder;
+
+import com.google.cloud.runtimes.builder.buildsteps.base.BuildStepException;
+import com.google.cloud.runtimes.builder.exception.AppYamlNotFoundException;
+import com.google.cloud.runtimes.builder.injection.RootModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Top-level class for executing from the command line.
+ */
+public class Application {
+
+  private static final Options CLI_OPTIONS = new Options();
+  private static final String EXECUTABLE_NAME = "<BUILDER_JAR>";
+
+  static {
+    CLI_OPTIONS.addRequiredOption("j", "jar-runtime", true, "base runtime to use for jars");
+    CLI_OPTIONS.addRequiredOption("s", "server-runtime", true, "base runtime to use for web server "
+        + "deployments");
+  }
+
+  /**
+   * Main method for invocation from the command line. Handles parsing of command line options.
+   */
+  public static void main(String[] args) {
+    CommandLine cmd = parse(args);
+    String jarRuntime = cmd.getOptionValue("j");
+    String serverRuntime = cmd.getOptionValue("s");
+
+    Path workspaceDir  = Paths.get(System.getProperty("user.dir"));
+
+    // Perform dependency injection and run the application
```

Comment:
I would put all of this code below this line into a separate static method: `build(jarRuntime, serverRuntime, workspaceDir)`. This will enable the class to be used more like a library without a CLI.